File: Data Structures & Algorithms/trapping-rain-water/submission-1.java
class Solution {
    public int trap(int[] height) {
        int l=0, r=height.length-1;
        int maxLeft=0, maxRight=0;
        int res=0;
        while(l<=r){
            if(maxLeft<=maxRight){
                int area = maxLeft-height[l];
                if(area>0){
                    res+=area;
                }
                maxLeft = Math.max(maxLeft, height[l]);
                l++;
            }else{
                int area = maxRight-height[r];
                if(area>0){
                    res+=area;
                }
                maxRight = Math.max(maxRight, height[r]);
                r--;
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/sum-of-two-integers/submission-1.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        res = 0
        while b:
            # tmp = a
            # a = a^b
            # b = (tmp&b) << 1
            tmp = (a&b)<<1
            a = a^b
            b=tmp
        return a


